crates/uv-cache/src/lib.rs

```diff
@@ -579,17 +602,28 @@ impl Cache {
                         continue;
                     }
 
-                    // Identify entries that match the wheel stem pattern (e.g., `typing-extensions-4.8.0-py3-none-any`).
                     let Some(filename) = entry
                         .path()
                         .file_name()
                         .and_then(|file_name| file_name.to_str())
                     else {
                         continue;
                     };
-                    if WheelFilename::from_stem(filename).is_err() {
+
+                    if bucket == CacheBucket::Wheels {
+                        // In `wheels` bucket hash is used for the directory name so we can't rely on stem pattern.
+                        // Instead we skip if it contains an extension (example: .whl, .http, .rev and .msgpack files).
+                        if filename
+                            .rsplit_once('-') // strip version/tags, might contain a dot ('.')
+                            .is_none_or(|(_, suffix)| suffix.contains('.'))
+                        {
+                            continue;
+                        }
+                    } else if WheelFilename::from_stem(filename).is_err() {
+                        // For other buckets only include entries that match the wheel stem pattern (e.g., `typing-extensions-4.8.0-py3-none-any`).
                         continue;
                     }
```

Comment:
I'd probably move this to a utility since it's non-trivial and repeated three times.

Reply:
(doesn't need to be blocking, we can do this separately)

Reply:
:+1: Moved the reference finding part to dedicated function. 

[`b400c85` (#11738)](https://github.com/astral-sh/uv/pull/11738/commits/b400c8598ddc19e347a683810027b2efa339f7b4#diff-d0c8455b65232353aa60383cd8a80d99a8b31cf7cd76bf22c18d32de36bed34cR514)